CenterForOpenScience/osf.io
# [IN-14] API Management of Moderators

## Purpose
Allow moderators to manage themselves, rather than requiring an OSF admin to add/remove moderators.

## Changes
* API endpoints (and all that's associated with them -- perms, serializers, etc.)
    * List / Create: `/v2/preprint_providers/<provider_id>/moderators/` 
    * Detail / Update / Delete: `/v2/preprint_providers/<provider_id>/moderators/<user_guid>/`
* Tests
* Email newly-added moderators (both existing users and when admins add via email)


## QA Notes
* No UI changes.
* TCI test permissions / basic behaviors, so QA time is likely not needed. However, if it is:
    * Admins can add/remove mods/admins (but not last admin)
    * Mods can see mods and remove themselves
    * Non-mods are not allowed to use these endpoints

## Side Effects
None expected. This does, however, allow the creation of (unregistered) users through API, which has some implications. However, it's very restrictive; only moderation admins can use it.

## Ticket
[[IN-14]](https://openscience.atlassian.net/browse/IN-14)
